hwtracer/src/perf/collect.c

```diff
@@ -270,24 +270,13 @@ bool read_aux(void *aux_buf, struct perf_event_mmap_page *hdr,
     new_data_size = (size - tail) + head;
   }
 
-  // Reallocate the trace storage buffer if more space is required.
   __u64 required_capacity = trace->len + new_data_size;
   if (required_capacity > trace->capacity) {
-    // Over-allocate to 2x what we need, checking that the result fits in
-    // the size_t argument of realloc(3).
-    if (required_capacity >= SIZE_MAX / 2) {
-      // We would overflow the size_t argument of realloc(3).
-      hwt_set_cerr(err, hwt_cerror_errno, ENOMEM);
-      return false;
-    }
-    size_t new_capacity = required_capacity * 2;
-    void *new_buf = realloc(trace->buf.p, new_capacity);
-    if (new_buf == NULL) {
-      hwt_set_cerr(err, hwt_cerror_errno, errno);
-      return false;
-    }
-    trace->capacity = new_capacity;
-    trace->buf.p = new_buf;
+    // FIXME: Reallocate the trace storage buffer if more space is required.
```

Comment:
Is this even a FIXME? Based on our discussion earlier, we may never want to reallocate?

Reply:
Agreed, we *probably* don't, but until we've measured things, I'm not totally sure.

Reply:
We can leave it a FIXME for now I guess.

We didn't adjust the buffer size. Do we want to do that either here or in a follow up PR?

Reply:
> We didn't adjust the buffer size. Do we want to do that either here or in a follow up PR?

Not sure I understand this.

Reply:
What I mean is, we no longer allow the buffer to grow, so is the initial buffer size appropriate.

I forget where the initial buffer size comes from. Does the user select it, or did we hard-code it?

Reply:
In practise it's set by constants in the C code, though users can override this.

Reply:
OK, let's get this in. We can always tweak it later.